src/main/java/seedu/smarthomebot/logic/commands/OnCommand.java

```diff
@@ -86,52 +108,40 @@ private String setParameter(String parameter, Appliance appliance) {
         return setParameterStatement;
     }
 
-    private CommandResult onByLocation() {
+    private CommandResult onByLocation() throws NoParameterForLocationException {
         if (!parameter.isEmpty()) {
-            return new CommandResult(MESSAGE_NO_PARAMETER_IN_ON_BY_LOCATION);
-        } else {
-            String outputResults = LINE;
-            outputResults = onApplianceByLoop(outputResults);
-            return new CommandResult(outputResults);
-        }
-    }
-
-    private CommandResult onByAppliance() {
-        int toOnApplianceIndex = getApplianceToOnIndex();
-        if (toOnApplianceIndex < 0) {
-            return new CommandResult(MESSAGE_APPLIANCE_OR_LOCATION_NOT_EXIST);
+            throw new NoParameterForLocationException();
         } else {
-            Appliance toOnAppliance = applianceList.getAppliance(toOnApplianceIndex);
-            String outputResult = onAppliance(toOnAppliance, "", false);
+            String outputResult = LINE;
+            outputResult = onApplianceByLoop(outputResult);
             return new CommandResult(outputResult);
         }
     }
 
-    private String onApplianceByLoop(String outputResults) {
+    private String onApplianceByLoop(String outputResult) {
```

Comment:
Is the argument for this parameter useful? Or can we directly declare outputResults in the function and return LINE + "All appliance in \"" + argument + "\" are turned on."

P.S, you're missing a full stop

Reply:
I try to be consistent in all the function